CI: documentation checks fail on github actions

Step "Build documentation and check for warnings" fails (last confirmed on Jan 24, 2022 for commit bd26dc276ba549f5b691bb065f65a27b7a12e4d4) with 

```
make: *** [makefile:293: doc] Error 1
Error: Process completed with exit code 2.
```
https://github.com/DrylandEcology/SOILWAT2/runs/4925554260?check_suite_focus=true


no other information is provided

doxygen v1.8.17 is installed

Creating doxygen documentation works without warnings/errors on my local setup (with doxygen v1.9.3)


-> enable this check under branch protection once this issue is resolved

--- a/SW_Flow_lib.c
+++ b/SW_Flow_lib.c
@@ -1797,25 +1797,26 @@ temp += temp;
 The algorithm selects a shorter time step if required for a stable solution
 (@cite Parton1978, @cite Parton1984).
 
-@param ptr_dTime Yesterday's successful time step in seconds.
-@param deltaX The depth increment for the soil temperature (regression) calculations (cm).
-@param sT1 The soil surface temperature as upper boundary condition (&deg;C).
-@param sTconst The soil temperature at a soil depth where it stays constant as
+@param[in,out] ptr_dTime Yesterday's successful time step in seconds.
+@param[in] deltaX The depth increment for the soil temperature (regression) calculations (cm).
+@param[in] sT1 The soil surface temperature as upper boundary condition (&deg;C).
+@param[in] sTconst The soil temperature at a soil depth where it stays constant as
 		lower boundary condition (&deg;C).
-@param nRgr The number of regressions (1 extra value is needed for the avgLyrTempR and oldavgLyrTempR for the last layer).
-@param avgLyrTempR An array of today's (regression)-layer soil temperature values (&deg;C).
-@param oldavgLyrTempR An array of yesterday's (regression)-layer soil temperature value (&deg;C).
-@param vwcR An array of temperature-layer VWC values (cm/layer).
-@param wpR An array of temperature-layer wilting point values (cm/layer).
-@param fcR An array of temperature-layer field capacity values (cm/layer).
-@param bDensityR temperature-layer bulk density of the whole soil
+@param[in] nRgr The number of regressions (1 extra value is needed for the avgLyrTempR and oldavgLyrTempR for the last layer).
+@param[in,out] avgLyrTempR An array of today's (regression)-layer soil temperature values (&deg;C).
+@param[in] oldavgLyrTempR An array of yesterday's (regression)-layer soil temperature value (&deg;C).
+@param[in] vwcR An array of temperature-layer VWC values (cm/layer).
+@param[in] wpR An array of temperature-layer wilting point values (cm/layer).
+@param[in] fcR An array of temperature-layer field capacity values (cm/layer).
+@param[in] bDensityR temperature-layer bulk density of the whole soil
   (g/cm<SUP>3</SUP>).
-@param csParam1 A constant for the soil thermal conductivity equation.
-@param csParam2 A constant for the soil thermal conductivity equation.
-@param shParam A constant for specific heat capacity equation.
-@param *ptr_stError A boolean indicating whether there was an error.
-@param surface_range Temperature range at the surface (&deg;C)
-@param temperatureRangeR An array of temperature ranges at each (regression)-layer to be interpolated (&deg;C)
+@param[in] csParam1 A constant for the soil thermal conductivity equation.
+@param[in] csParam2 A constant for the soil thermal conductivity equation.
+@param[in] shParam A constant for specific heat capacity equation.
+@param[in,out] *ptr_stError A boolean indicating whether there was an error.
+@param[in] surface_range Temperature range at the surface (&deg;C)
+@param[in,out] temperatureRangeR An array of temperature ranges at each (regression)-layer to be interpolated (&deg;C)
+@param[in] depthsR Evenly spaced depths of the soil temperature profile (cm).
 
 @note
 	avgLyrTempR[0] and temperatureRangeR[0] represent soil surface conditions.

--- a/doc/Doxyfile
+++ b/doc/Doxyfile
@@ -1620,7 +1620,7 @@ FORMULA_MACROFILE      =
 # The default value is: NO.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-USE_MATHJAX            = NO
+USE_MATHJAX            = YES
 
 # With MATHJAX_VERSION it is possible to specify the MathJax version to be used.
 # Note that the different versions of MathJax have different requirements with

--- a/doc/doxygen_exceptions.txt
+++ b/doc/doxygen_exceptions.txt
@@ -1,1 +1,2 @@
-
+SOILWAT2/NEWS.md:[0-9]\+: warning: Found unknown command
+SOILWAT2/README.md:[0-9]\+: warning: unable to resolve reference to

--- a/rands.c
+++ b/rands.c
@@ -388,7 +388,7 @@ double RandNorm(double mean, double stddev, pcg32_random_t* pcg_rng) {
 
   The beta distribution has two shape parameters \f$a\f$ and \f$b\f$.
   The density is
-      \f[\frac{x ^ (a - 1) * (1 - x) ^ (b - 1)}{Beta(a, b)}\f]
+      \f[\frac{x ^ {(a - 1)} * (1 - x) ^ {(b - 1)}}{Beta(a, b)}\f]
   for \f$0 < x < 1\f$
 
 	The code for RandBeta was taken from ranlib, a FORTRAN77 library. Original

--- a/tools/run_doxygen.sh
+++ b/tools/run_doxygen.sh
@@ -32,14 +32,19 @@ doxygen ${doxy} > ${log} 2>&1
 # Make sure that there is such a (readable) file
 if [ -r ${doxexcept} ]; then
   # Make sure that there are exceptions in the file
-  if [ $(wc -l < ${doxexcept}) -ne 0 ]; then
+  if [ $(wc -w < ${doxexcept}) -ne 0 ]; then
     # Remove exceptions from the logfile
-    grep -v -f ${doxexcept} ${log} > ${log_tmp}
+    grep --invert-match -f ${doxexcept} ${log} > ${log_tmp}
     mv ${log_tmp} ${log}
     rm -f ${log_tmp}
   fi
 fi
 
+# Fail if log file is completely empty (likely a grep-parsing error)
+if [ $(wc -w < ${log}) -eq 0 ]; then
+  echo "doxygen log file is empty pointing to a likely error."
+  exit 1
+fi
 
 # Examine log file for remaining warnings/errors
 warnings="$(grep -iE "warning|error" ${log} 2>&1 || echo "")"

